sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -1052,6 +1052,13 @@ object RapidsConf {
     .booleanConf
     .createWithDefault(true)
 
+  val REGEXP_MAX_STATE_MEMORY = conf("spark.rapids.sql.regexp.maxStateMemory")
+    .doc("Specifies the maximum memory on GPU to be used for regular expressions." +
+      "Note that the memory usage is an estimate based on a worst case using only " +
```

Comment:
I am not sure how I feel about calling this worst case. Because it kind of is a worst case, except the worst case for the number of rows could be much larger than what you use because we estimate the number of rows.  Perhaps we can clarify this a little.